aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/SubscriptionAuthorizer.java

```diff
@@ -63,58 +69,45 @@ JSONObject getAuthorizationDetails(boolean connectionFlag,
+            authMode = mAwsConfiguration.optJsonObject("AppSync").getString("AuthMode");
         } catch (JSONException e) {
             throw new RuntimeException("Failed to read AuthMode from awsconfiguration.json", e);
         }
 
         // Construct the Json based on the Auth Mode
         switch (authMode) {
             case "API_KEY" :
-                return getAuthorizationDetailsForApiKey(awsConfiguration);
+                return getAuthorizationDetailsForApiKey();
             case "AWS_IAM" :
-                return getAuthorizationDetailsForIAM(connectionFlag, awsConfiguration,
-                        subscription, applicationContext);
+                return getAuthorizationDetailsForIAM(connectionFlag, subscription);
             case "AMAZON_COGNITO_USER_POOLS" :
-                return getAuthorizationDetailsForUserpools(awsConfiguration, applicationContext);
+                return getAuthorizationDetailsForUserpools();
             case "OPENID_CONNECT" :
-                return getAuthorizationDetailsForOidc(awsConfiguration, mOidcAuthProvider);
+                return getAuthorizationDetailsForOidc();
             default :
                 throw new RuntimeException("Invalid AuthMode read from awsconfiguration.json.");
         }
     }
 
-    private static JSONObject getAuthorizationDetailsForApiKey(AWSConfiguration awsConfiguration) {
+    private JSONObject getAuthorizationDetailsForApiKey() {
         try {
             return new JSONObject()
-                .put("host", getHost(getApiUrl(awsConfiguration)))
+                .put("host", getHost(getApiUrl()))
                 .put("x-amz-date", ISO8601Timestamp.now())
-                .put("x-api-key", awsConfiguration.optJsonObject("AppSync").getString("ApiKey"));
+                .put("x-api-key", getApiKey());
         } catch (JSONException | MalformedURLException e) {
             throw new RuntimeException("Error constructing the authorization json for Api key. ", e);
         }
     }
 
-    private static JSONObject getAuthorizationDetailsForIAM(boolean connectionFlag, AWSConfiguration awsConfiguration,
-                                                            Subscription subscription,
-                                                            Context applicationContext) throws JSONException {
-        String identityPoolId;
-        String regionStr;
-        try {
-            JSONObject identityPoolJSON = awsConfiguration.optJsonObject("CredentialsProvider")
-                .getJSONObject("CognitoIdentity")
-                .getJSONObject(awsConfiguration.getConfiguration());
-            identityPoolId = identityPoolJSON.getString("PoolId");
-            regionStr = identityPoolJSON.getString("Region");
-        } catch (JSONException e) {
-            throw new RuntimeException("Error reading identity pool information from awsconfiguration.json", e);
-        }
```

Comment:
These have been moved into [`getCredentialsProvider`](https://github.com/awslabs/aws-mobile-appsync-sdk-android/pull/276/files#diff-c7b10e562c4683fe927f8ade5e7121d0R166) since they were only used to create a credentials provider later on.